src/components/download-track-progress-drawer/DownloadTrackProgressDrawer.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useCallback } from 'react'
+
+import LottieView from 'lottie-react-native'
+import { Platform, StyleSheet, View } from 'react-native'
+import { useSelector } from 'react-redux'
+
+import Drawer from 'app/components/drawer'
+import Text from 'app/components/text'
+import { useDrawer } from 'app/hooks/useDrawer'
+import { useThemedStyles } from 'app/hooks/useThemedStyles'
+import {
+  getDownloadedPercentage,
+  getFileName,
+  getFetchTask
+} from 'app/store/download/selectors'
+import { useColor, ThemeColors } from 'app/utils/theme'
+
+const createStyles = (themeColors: ThemeColors) =>
+  StyleSheet.create({
+    view: {
+      display: 'flex',
+      flexDirection: 'column',
+      alignItems: 'center',
+      paddingTop: 16,
+      paddingHorizontal: 24,
+      fontFamily: 'AvenirNextLTPro-Regular'
+    },
+
+    title: {
+      fontSize: 21,
+      color: themeColors.actionSheetText,
+      marginBottom: 28
+    },
+
+    fileName: {
+      fontSize: 18,
+      lineHeight: 24,
+      color: themeColors.neutralLight4,
+      marginBottom: 12,
+      textAlign: 'center'
+    },
+
+    downloadContainer: {
+      display: 'flex',
+      flexDirection: 'row',
+      alignItems: 'center',
+      marginBottom: 32
+    },
+
+    loadingIcon: {
+      width: 40,
+      height: 40,
+      alignSelf: 'flex-start',
+      marginRight: 10
+    },
+
+    downloadPercent: {
+      fontSize: 40,
+      color: themeColors.neutral,
+      lineHeight: 46
+    }
+  })
+
+const messages = {
+  title: 'Downloading'
+}
+
+const DownloadTrackProgressDrawer = () => {
+  const [isOpen, setIsOpen] = useDrawer('DownloadTrackProgress')
+
+  const downloadPercentage = useSelector(getDownloadedPercentage)
+  const fetchTask = useSelector(getFetchTask)
+  const fileName = useSelector(getFileName)
+
+  const handleClose = useCallback(() => {
+    if (fetchTask) {
+      fetchTask.cancel()
+    }
+    setIsOpen(false)
+  }, [fetchTask, setIsOpen])
+
+  const styles = useThemedStyles(createStyles)
+  const spinnerColor = useColor('actionSheetText')
+
+  return (
+    <Drawer isOpen={isOpen} onClose={handleClose}>
+      <View style={styles.view}>
+        <Text style={styles.title} weight='bold'>
+          {messages.title}
+        </Text>
+        <Text style={styles.fileName} weight='bold'>
+          {fileName}
+        </Text>
+        <View style={styles.downloadContainer}>
+          <LottieView
+            style={styles.loadingIcon}
```

Comment:
probably worth extracting into a shared component for re-use. no action here but figured i'd flag.